docs/tuning.md

```diff
@@ -232,8 +232,12 @@ temporary objects created during task execution. Some steps which may be useful
   value of the JVM's `NewRatio` parameter. Many JVMs default this to 2, meaning that the Old generation 
   occupies 2/3 of the heap. It should be large enough such that this fraction exceeds `spark.memory.fraction`.
   
-* Since 4.0.0, Spark uses JDK 17 by default, which also makes the G1GC garbage collector the default. Note that with
-  large executor heap sizes, it may be important to increase the G1 region size with `-XX:G1HeapRegionSize`.
+* Since 4.0.0, Spark uses JDK 17 by default, which also makes the G1GC garbage collector the default,
+  Some applications may encounter out-of-memory (OOM) issues or experience slower performance.
+  To mitigate these problems:
+  - Increase region size with `-XX:G1HeapRegionSize=<size>` when using large executor heap sizes.
+  - Optimize GC locker behavior with `-XX:+UnlockDiagnosticVMOptions -XX:GCLockerRetryAllocationCount=100`.
```

Comment:
Might be worth mentioning that the purpose of this suggestion is to mitigate https://openjdk.org/jeps/423, and is not required when running Spark with JDK 22+

PS: I have verified some internal Spark workloads, which were easy to OOM with JDK 21 and G1, that work well on JDK 22. Unfortunately, JDK 22 is EOL, life will be easier if the openJDK community can backport JEP 423 to JDK 21 LTS